src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Stores/OpenIdAuthorizationStore.cs

```diff
@@ -359,12 +370,12 @@ public virtual async ValueTask PruneAsync(CancellationToken cancellationToken)
                 cancellationToken.ThrowIfCancellationRequested();
 
                 var authorizations = await _session.Query<TAuthorization, OpenIdAuthorizationIndex>(
-                    authorization => authorization.Status != OpenIddictConstants.Statuses.Valid ||
-                   (authorization.Type == OpenIddictConstants.AuthorizationTypes.AdHoc &&
-                    authorization.AuthorizationId.IsNotIn<OpenIdTokenIndex>(
-                        token => token.AuthorizationId,
-                        token => token.Status == OpenIddictConstants.Statuses.Valid &&
-                                 token.ExpirationDate > DateTimeOffset.UtcNow))).Skip(offset).Take(1_000).ListAsync();
+                    authorization => authorization.CreationDate < threshold &&
+                                    (authorization.Status != OpenIddictConstants.Statuses.Valid ||
+                                    (authorization.Type == OpenIddictConstants.AuthorizationTypes.AdHoc &&
+                                     authorization.AuthorizationId.IsNotIn<OpenIdTokenIndex>(
+                                         token => token.AuthorizationId,
+                                         token => token.Id != 0)))).Skip(offset).Take(1_000).ListAsync();
```

Comment:
@sebastienros I'm not a huge fan of this workaround. Do you think it would make sense to have an `IsNotIn` overload that doesn't require a second argument? Or am I using this stuff incorrectly? :confused:

Reply:
Didn't think about it. Very easy to handle an overload that would not have a constraint.

Code is here:

https://github.com/sebastienros/yessql/blob/2e4d526cff556dd41ec108e3837b59c94bd5c968/src/YesSql.Core/Services/DefaultQuery.cs#L254-L255

 You can file an issue. You can even just return `true` I assume. That should generate `WHERE 1` :/

Reply:
> You can file an issue.

Will do!